### PR TITLE
Completely remove the capsule and cross

### PR DESCRIPTION
**Pull request by mohanson**
- Completely remove the capsule and cross. 
- Upgraded clang to clang-18. Since capsule and its docker environment is removed, you must install clang-18 or riscv64-unknown-elf-gcc locally.

### CODE REVIEW

File: contracts/spawn-caller/Cargo.toml [hunk 1]
```diff
@@ -6,4 +6,4 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-ckb-std = { path = "../../" }
+ckb-std = { path = "../../", features=["build-with-clang", "dummy-atomic"] }
```
Comment (XuJiandong) [inline contracts/spawn-caller/Cargo.toml]: Is "dummy-atomic" mandatory or optional?

File: test/Makefile [hunk 2]
```diff
@@ -6,8 +6,7 @@ test: build
 
 build:
 	make -C shared-lib all-via-docker
-	cd .. && capsule build
-	cd .. && RUSTFLAGS="-C target-feature=-a" capsule build -n ckb-std-tests
+	cd ../contracts && RUSTFLAGS="-C target-feature=-a" cargo build --target riscv64imac-unknown-none-elf
```
Comment (XuJiandong) [inline test/Makefile:9]: Build it as "release". debug version is bloat.

File: test/src/contract.rs [hunk 3]
```diff
@@ -14,7 +14,7 @@ fn it_works() {
     let mut context = Context::default();
     let contract_bin = {
         let mut buf = Vec::new();
-        File::open("../build/debug/ckb-std-tests")
+        File::open("../contracts/target/riscv64imac-unknown-none-elf/debug/ckb-std-tests")
```
Comment (XuJiandong) [inline test/src/contract.rs:17]: Update path from "debug" to "release"

**Review by XuJiandong — CHANGES_REQUESTED** [review]

File: contracts/spawn-caller/Cargo.toml (hunk 1 above)
Comment (mohanson) [reply contracts/spawn-caller/Cargo.toml]: This is required. When using ckb-gen-types >= 0.117, the output will include the A instructions.

Comment (XuJiandong) [reply contracts/spawn-caller/Cargo.toml]: So it's better to set "dummy-atomic" as default feature.

Comment (XuJiandong) [reply contracts/spawn-caller/Cargo.toml]: When "dummy-atomic"  is set as default feature, bump major version (0.16).

**Review by XuJiandong — CHANGES_REQUESTED** [review]

**Review by XuJiandong — APPROVED** [review]

**Comment by XuJiandong** [comment]
fix warnings:
```
warning: unused variable: `err`
   --> /Users/xxx/projects/ckb-std/src/high_level.rs:569:17
    |
569 |             Err(err) => {
    |                 ^^^ help: if this is intentional, prefix it with an underscore: `_err`
    |
    = note: `#[warn(unused_variables)]` on by default

warning: `ckb-std` (lib) generated 1 warning (run `cargo fix --lib -p ckb-std` to apply 1 suggestion)
   Compiling simulator v0.1.0 (/Users/xxx/projects/ckb-std/test/simulator)
warning: unused variable: `v`
  --> src/../../../contracts/ckb-std-tests/src/entry.rs:34:9
   |
34 |     let v = vec![0u8; 42];
   |         ^ help: if this is intentional, prefix it with an underscore: `_v`
   |
   = note: `#[warn(unused_variables)]` on by default

warning: unused variable: `data_hash`
   --> src/../../../contracts/ckb-std-tests/src/entry.rs:157:9
    |
157 |     let data_hash = packed::CellOutput::calc_data_hash(&data);
    |         ^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_data_hash`

warning: `simulator` (bin "contract") generated 2 warnings (run `cargo fix --bin "contract"` to apply 2 suggestions)
    Finished dev [unoptimized + debuginfo] target(s) in 1m 00s
```

**Comment by mohanson** [comment]
These warnings have been there since these PRs:

https://github.com/nervosnetwork/ckb-std/commit/e544f02b5ed4146b63b9a48c0e094941a6edd9ff

https://github.com/nervosnetwork/ckb-std/commit/22648e2642a3d784cbb0bd31fc27275cdd239b65

**Pull request merged**